### ISSUE TITLE
[JAVA] RxJava2 support 

### ISSUE THREAD

**Issue by ber4444** [swagger-api/swagger-codegen#4698]
##### Description

When using the configuration settings, 
```
  "library": "retrofit2",
  "useRxJava": true,
```
we are getting these dependencies:
```
ext {
    retrofit_version = "2.0.2"
    rx_java_version = "1.1.3"
}
dependencies {
    compile "com.squareup.retrofit2:retrofit:$retrofit_version"
    compile "com.squareup.retrofit2:converter-gson:$retrofit_version"
    compile "com.squareup.retrofit2:adapter-rxjava:$retrofit_version"
    compile "io.reactivex:rxjava:$rx_java_version"
```
However, we would like to add a useRxJava2 setting which would result in:
```
ext {
    retrofit_version = "2.0.2"
    rx_java_version = "2.0.1"
}
dependencies {
    compile "com.squareup.retrofit2:retrofit:$retrofitVersion"
    compile "com.squareup.retrofit2:converter-gson:$retrofitVersion"
    compile "com.jakewharton.retrofit:retrofit2-rxjava2-adapter:1.0.0"
    compile "io.reactivex.rxjava2:rxjava:$rx_java_version"
```

##### Related issues

Similar to this PR:
https://github.com/swagger-api/swagger-codegen/pull/1928/files

**Comment by cbornet** [comment]
Would you do a PR for that ?

**Comment by wing328** [comment]
Another option is to customize the template: https://github.com/swagger-api/swagger-codegen/blob/1dd9ee39af6c4e2f9c016a94d16ca47a88a9cb0d/modules/swagger-codegen/src/main/resources/Java/libraries/retrofit2/build.gradle.mustache#L102 and then use the `-t` option with the customized templates.

**Comment by ber4444** [comment]
I'll do the PR. Not as simple as @wing328 suggests because the class and package names changed as well.

**Comment by wing328** [comment]
@ber4444 thanks! I think we'll need to provide a CLI option, e.g. useRxJava2, so that users still have the option to use RxJava 1.x. Let me know if you need help on that.

**Comment by wing328** [comment]
@ber4444 I'll review your PR and let you know if I've any question.

**Comment by ber4444** [comment]
I updated the PR based on your feedback.

**Comment by wing328** [comment]
@ber4444 I've found 2 minor issues: https://github.com/swagger-api/swagger-codegen/issues/4708. Let me know if you need help fixing it.

**Comment by wing328** [comment]
Closed via #4743 

Thanks for the PR from @ber4444

**Comment by adhigadu** [comment]
hello @wing328 @ber4444  I was having similar issue and would like to use rxjava2 with retrofit. Can you please confirm if its already in 2.2.2 or will be released on 2.2.3. If its in 2.2.3 would i be able to access pre-release 2.2.3 jar file please.. thanks..
